.cursor/rules/frontend_development.mdc

```diff
@@ -0,0 +1,432 @@
+---
+description: "Guidelines for frontend development in Langflow, focusing on React/TypeScript UI components, build processes, and frontend testing."
+globs:
+  - "src/frontend/**/*.{ts,tsx,js,jsx}"
+  - "src/frontend/**/*.{css,scss,json}"
+  - "src/frontend/package*.json"
+  - "src/frontend/vite.config.*"
+  - "src/frontend/tailwind.config.*"
+  - "src/frontend/tsconfig.json"
+alwaysApply: false
+---
+
+
+# Frontend Development Guidelines
+
+## Purpose
+Guidelines for frontend development in Langflow, focusing on React/TypeScript UI components, build processes, and frontend testing.
+
+---
+
+## 1. Frontend Environment Setup
+
+### Prerequisites
+- **Node.js:** v22.12 LTS for JavaScript runtime
+- **Package Manager:** npm (v10.9) for dependency management
+- **Development Tools:** Vite for build tooling
+
+### Frontend Service
+```bash
+make frontend  # Start Vite dev server on port 3000
+```
+- Hot-reload enabled for UI changes
+- Access at: http://localhost:3000/
+- Frontend source: `src/frontend/`
+
+---
+
+## 2. Frontend Structure
+
+### Directory Layout
+```
+src/frontend/src/
+├── components/          # Reusable UI components
+├── pages/              # Page-level components
+├── icons/              # Component icons and lazy loading
+├── stores/             # State management (Zustand)
+├── types/              # TypeScript type definitions
+├── utils/              # Utility functions
+├── hooks/              # Custom React hooks
+├── services/           # API service functions
+└── assets/             # Static assets
+```
+
+### Key Technologies
+- **React 18** with TypeScript
+- **Vite** for build tooling and dev server
+- **Tailwind CSS** for styling
+- **Zustand** for state management
+- **React Flow** for flow graph visualization
+- **Lucide React** for icons
+
+---
+
+## 3. Component Icon Development
+
+### Adding Component Icons
+1. **Create Icon Directory:**
```

Comment:
Steps 1-4 seem to be duplicates of the `icons.mdc` should we just reference that?

Reply:
Good catch. Removed it.